javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -103,13 +107,9 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
      * @throws NullPointerException if {@code elements} is null
      */
     @SafeVarargs
+    @SuppressWarnings("varargs")
     public static <T> Vector<T> of(T... elements) {
-        Objects.requireNonNull(elements, "elements is null");
-        HashArrayMappedTrie<Integer, T> result = HashArrayMappedTrie.empty();
-        for (T element : elements) {
-            result = result.put(result.size(), element);
-        }
-        return wrap(result);
+        return ofAll(List.of(elements));
```

Comment:
I had to find an intermediary representation that removed the `HAMT` (but was still working correctly).
In a later commit it's replaced again with `BMT`, but will replace it with `appendAll` or something here :)